threatingestor/sources/github.py

```diff
@@ -8,29 +8,56 @@
 SEARCH_URL = "https://api.github.com/search/repositories"
 
 class Plugin(Source):
-
-    def __init__(self, name, search):
+    """Github Source Plugin"""
+    def __init__(self, name, search, username="", token=""):
         self.name = name
         self.search = search
 
+        if username and token:
+            self.auth = (username, token)
+        else:
+            self.auth = None
+
+    def make_requests(self, url, params, auth):
```

Comment:
Can we rename this method to something more specific, and add a leading underscore to show it's for internal use only? Maybe `_repository_search` or something?